packages/pluggableWidgets/rich-text-web/src/RichText.tsx

```diff
@@ -0,0 +1,75 @@
+import { ReactNode, createElement, useCallback } from "react";
+import { RichTextEditor as RichTextComponent } from "./components/RichText";
+import { RichTextContainerProps } from "../typings/RichTextProps";
+import { GroupType, SET_CUSTOM } from "./utils/ckeditorPresets";
+import { debounce } from "@mendix/piw-utils-internal";
+import { getPreset, defineAdvancedGroups } from "./utils/ckeditorConfigs";
+import { CKEditorConfig } from "ckeditor4-react";
+import "./ui/RichText.scss";
+
+export default function RichText(props: RichTextContainerProps): ReactNode {
+    const onChangeFn = useCallback(
+        debounce((value: string) => props.stringAttribute.setValue(value), 500),
+        [props.stringAttribute]
+    );
+    const defineToolbar = (): CKEditorConfig => {
+        const { advancedConfig, toolbarConfig, preset } = props;
+        if (preset !== "custom") {
+            return getPreset(preset);
+        } else {
+            const groupKeys = Object.keys(props).filter((key: string) => (key.includes("Group") ? key : null));
```

Comment:
I will prefer to keep it dynamic, making static list of item is a bad idea, cause in different cases the number of items can be much more for example.

`groupKeys` now just making sure that you are passing correct group names to 'createCustomToolbar' method. But in general, if you add something which is not in the `ToolbarItems` it will be excluded from final list.

Reply:
> cause in different cases the number of items can be much more for example.

What do you mean exactly and how does that make a static list bad?

> it will be excluded from final list.

Can you point me to the code that does that? Because right now I only see this filtering, which only checks for "Group".

Reply:
My point is it's better to get any props dynamically if you can, rather than hardcoding the staff.

And here is the part when I am checking if the item is in the list
[packages/pluggableWidgets/rich-text-web/src/utils/ckeditorPresets.ts](https://github.com/mendix/widgets-resources/pull/1125/files#diff-1f1184f8a3abe3737a7db00c3403ec989aa59ef2dcb07ba2951a516e391741b9R124)

In any case, if in the future, someone by mistake add another parameter which includes "Group" in the name, it will be excluded from final list.